### ISSUE TITLE
contactnotifications queries might cause CPU load waiting for notifications insert ID

### ISSUE THREAD

**Issue by dnsmichi** [Icinga/icinga2#5024]
### Problem

The problem is that the contactnotifications queries are rescheduled quite often until the `notifications` table insert id is available.

### Analysis

The culprit here is that the insert does not take place immediately but has to wait until the client multi statement is flushed and then the insert id is available.

If the query queue is literally empty, this might cause some cpu cycles with re-enqueuing until FinishAsyncQueries() is called. 

This is a not noticeable issue on large scale setups which always hit the 25k query limit quite easily.

I'm leaving this issue as notice for anyone looking into cpu spikes when dumping notifications into the IDO database (maybe on embedded hardware).

The real problem is the way how notifications are handled in the legacy IDO schema. Depending on a runtime insert id makes it impossible to 

- use multiple connections for asynchronous queries
- just insert the notified users and notifications independent from each other

Basically it is a won't fix, but if anyone comes up with a solution in the next months, much appreciated.

The analysis sources from #4603.

### Logs

**Note**: You don't see those many debug log entries using a release package. That is only available in debug builds.

Log is too big for the 64k body limit GitHub enforces. Attached.

[ido_contactnotifications_waiting_for_notification_insert_id.txt](https://github.com/Icinga/icinga2/files/796257/ido_contactnotifications_waiting_for_notification_insert_id.txt)

**Comment by dnsmichi** [comment]
This is a wont-fix for the current IDO backend.
